### DISCUSSION TITLE
Reconstructing all vectors with Arbitrary ID mapping

### DISCUSSION THREAD

**Discussion by Isaac-the-Man** [facebookresearch/faiss#3560] [Q&A]
# Summary

How do I reconstruct all vectors from an Index with ID mapping enabled? The IDs are non-contiguous arbitrary integers in my case, and calling `reconstruct_n(0, index.ntotal)` throws a Fatal Python Error which I assume is because faiss is reconstructing the vectors based on my non-contiguous ID mapping.

If I understand this correctly, I should be able to get pass the ID maps and call `reconstruct_n` directly on the Index, which I assume still uses incremental IDs starting at 0.

I'm aware that I can always loop through the IDs and call `reconstruct` on each item, but I believe there must be a better way?

# Platform

OS: 

Faiss version: 

Installed from:  

Faiss compilation options: 

Running on:
- [x] CPU
- [ ] GPU

Interface: 
- [ ] C++
- [x] Python

# Reproduction instructions

**Comment by mdouze** [comment]
Please use `reconstruct_batch` with the ids you want to reconstruct.

**Comment by Isaac-the-Man** [comment]
Thanks for the quick response, `reconstruct_batch` works perfectly for me! 

I'd still like to know if there is any way to bypass ID Mapping and call all the `reconstruct_x` methods directly on the default incremental ID?
